3p/integration-lib.js

```diff
@@ -272,14 +272,12 @@ export function validateAllowedEmbeddingOrigins(window, allowedHostnames) {
   // We prefer the unforgable ancestorOrigins, but referrer is better than
   // nothing.
   const ancestor = ancestors ? ancestors[0] : window.document.referrer;
-  let {hostname} = parseUrlDeprecated(ancestor);
+  let {hostname} = parseUrl(ancestor);
   if (isProxyOrigin(ancestor)) {
     // If we are on the cache domain, parse the source hostname from
     // the referrer. The referrer is used because it should be
     // trustable.
-    hostname = parseUrlDeprecated(
-      getSourceUrl(window.document.referrer)
-    ).hostname;
+    hostname = parseUrl(getSourceUrl(window.document.referrer)).hostname;
```

Comment:
nit because I happen to be seeing this now; this parses the url the first time even when it's unused.
```javascript
const url = isProxyOrigin(ancestor) ? ancestor : getSourceUrl(window.document.referrer);
let {hostname} = parseUrl(url);
```